mantidproject/mantid
# Fix some issues with EventWorkspace MRU

This fixes two correctness issues in the MRU of `EventWorkspace`. There is also some related cleanup and small refactoring to eliminate dependencies of `EventList` on the spectrum number, which is a prerequisite for #16810.

**To test:**
- Code review, in particular make sure that using the `this` pointer as MRU index is correct (or at least not less correct that the spectrum number).
- Check that the regression tests catch the issues if they are reintroduced.

Fixes #17061 
Fixed #17070 
No release notes, no know issues seen in the wild.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
